### ISSUE TITLE
How do I specify a "constants" filter with Excel style filtering ?

### ISSUE THREAD

**Issue by catull** [IgniteUI/igniteui-angular#6492]
## Question
I am aware of how to use Excel style filter mode.

I read the entire issue #3829, it is instructive.

However, it is useful to filter POST-fetching, I want to initialise the filter PRE-fetching, with a set of constants, say a list of countries.

I have initialised the filters.
But once the page is rendered, the filters are empty.

Is there a way to supply an array of values to the Excel filter mode as initial status ?

 * igniteui-angular version: 9.0.0.beta.6
 * browser: all

Thanks.

**Comment by nrobakova** [comment]
Hi @catull,
When you use  Excel style filter mode and want the filter to be applied in the Excel style filter you have to use the column FilteringExpressionsTree.
You cam look at this [link ](FilteringExpressionsTree) for more detail.
Here is a link to the StackBlitz  [sample ](https://stackblitz.com/edit/angular-ibfhun-gdygbh)similar to this one in the bug 3829, but with different setting of the FilteringExpressionsTree.

**Comment by catull** [comment]
@nrobakova

The first link is not useful, at all.  It does a look up of anything the author "" ever created, which is 0 issues.
Even when I filter the issues for ["FilteringExpressionTree",](https://github.com/IgniteUI/igniteui-angular/issues?utf8=%E2%9C%93&q=FilteringExpressionsTree) this is not helping either.
I looked at few but none of them address my concern.

I know how to use it and create `FilteringExpressionTree`, btw.

Also, the StackBlitz example is not helpful either.

I did not ask for an example with a separate combo-field.
That example is shown on the official page, as well.

I asked for a population of a fixed drop-down used as a filter of a  grid column.

**Comment by nrobakova** [comment]
@catull 

If you want to display specific values in the filter of a column you can set uniqueColumnValuesStrategy and to add values which you want. 

Can you send me a small sample or image how you are expecting filter of column to be previewed?

**Comment by catull** [comment]
@nrobakova Thanks.

I implemented `[uniqueColumnValuesStrategy]="columnValuesStrategy"` and
```typescript
  public columnValuesStrategy = (
    column: IgxColumnComponent,
    columnExprTree: IFilteringExpressionsTree,
    done: (uniqueValues: any[]) => void,
  ) => {
    const options: string[] = OPTIONS[column.field] || [];
    done(options);
    columnExprTree.fieldName; // have to leave this statement in because of strict compiler setting
  };
```
I have some observations.

0.  The arguments of `columnValuesStrategy` are better grouped into a ColumnValueStrategy of sorts.
    As this is not the case, I am forced to using both `column` and `columnExprTree` parameters.

1. This `uniqueColumnValuesStrategy` is for the whole grid.
     It is not ideal, since I have only 3 columns out of 8 which require constant values.
     It would be great if that functionality were available per column instead.

2. There does not seem to be a way to initialise the filter in `ngOnInit` or `ngAfterViewInit`.
    The `uniqueColumnValuesStrategy` is called as soon as I click on the inverse pyramid icon, long after `ngOnit` and `ngAfterViewInit` were called.

3.  How can I pre-select the values ?
     Once I pass back the 4 status value above, I want to set "OK" as selected, without requiring an user interaction.
     
In other words, if I navigate from a different component, and I pass along "&status=OK", I'd like the data fetched to be pre-selected.

This does not seem to be possible.

**Comment by nrobakova** [comment]
Hi @catull,
If you want you can add a new feature request uniqueColumnValuesStrategy to be per column, not per the whole grid.
You can still filter the grid in the in the ngAfterViewInit and the values from the set from the uniqueColumnValuesStrategy  will be selected. 
Here is a link to a StackBlitz [sample](https://stackblitz.com/edit/angular-ibfhun-kjj7ox?file=src/app/grid/grid-excel-style-filtering-sample-1/) with uniqueColumnValuesStrategy  and filter applied.

**Comment by zdrawku** [comment]
@catull, let us know if you have any questions
